src/clj/fluree/db/api.cljc

```diff
@@ -314,11 +315,13 @@
 
 (defn credential-query-connection
   ([conn cred-query] (credential-query-connection conn cred-query {}))
-  ([conn cred-query opts]
+  ([conn cred-query {:keys [format] :as opts}]
    (promise-wrap
     (go-try
-     (let [{query :subject, identity :did} (<? (cred/verify cred-query))]
-       (<? (query-connection conn query (assoc opts :did identity))))))))
+     (let [json? (not= :sparql format)
```

Comment:
Would it be possible to pass the actual format key down? I tend to prefer arguments that carry a bit of semantics with them, rather than a naked boolean.

Reply:
I think the issue here is that `cred-verify` should be a verifier and not a JSON parser. Those two are conflated here.

I'd rather split up those roles... but my goal here wasn't refactoring code, it was fixing what was not working.

Reply:
And by that I also mean that fluree/server relies on cred/verify and it's JSON parsing... so any change in that behavior is going to affect not just fluree/db/

Reply:
I think it makes sense for us to change that behavior sooner rather than later because quick short term changes to get things working have a tendency to be codified into the api.

Reply:
Made a ticket here: https://github.com/fluree/db/issues/845

Reply:
> I think it makes sense for us to change that behavior sooner rather than later because quick short term changes to get things working have a tendency to be codified into the api.

The function already did JSON parsing, that is not new. What is new is that a new arity option can tell it to not do that.

Reply:
Right. What I meant was that we should separate parsing from verification sooner rather than later if that's what we wanted to do. 

I'm not the biggest fan of the boolean argument either. If we were to keep the parsing here long term, then I would support instead splitting the functions up in a different way. Instead of a `verify` function with a `json?` argument, I think it would be better to have two functions: `verify-json` and `verify-string`, and the function here would call one of them depending on whether the format was `:sparql` or not.

Reply:
Not sure why CI/CD stopped working - bunch of tests are now failing so pulling this down for now. Maybe re-enabling it after I figure out the failures will kick it off again.

Reply:
Hopefully addressed this. Changed the following:
1) removed JSON option that was added
2) remove JSON parsing from `cred/verify-jws`, and moved to `cred/verify` only for JWS
3) Refactored credential APIs for SPARQL to call `cred/verify-jws` directly, and no longer use `cred/verify`

In summary, `cred/verify` functionality is identical to that before and everything relying on it will remain working, but for SPARQL we now no longer use that function (and therefore avoid needing adding the extra JSON parsing flag)

Reply:
Updated https://github.com/fluree/db/issues/845 to reflect this latest set of changes, removing options of how to handle boolean flag because it is no longer being used.